docs/developer-guide/running-substreams.md

```diff
@@ -77,3 +77,36 @@ map_transfers: message "eth.erc721.v1.Transfers": {
 ```
 
 The example output shown above contains data for different transfers from data in the blockchain. These transfers can also be verified on [Etherscan](https://etherscan.io/tx/0xcfb197f62ec5c7f0e71a11ec0c4a0e394a3aa41db5386e85526f86c84b3f2796).
+
+## Development & Production Mode
+
+### Development Mode
+
+Production and development modes impact the execution of Substreams. Key aspects of execution include:
+
+- The time required to reach the first byte.
+- The speed that large ranges get executed.
+- The module logs and outputs that are sent back to the client.
+
+Differences between production and development modes include:
+
+- Forward execution is enabled in production mode and disabled in development mode.
+- The time required to reach the first byte in development mode is faster than in production mode.
+- In development mode, the client will receive all of the executed module's logs. In production mode the client will only receive the output module's logs.
+- In development mode, it's possible to request specific store snapshots in the execution tree.
+
+Examples: (given the dependencies: `[block] --> [map_pools] --> [store_pools] --> [map_transfers])`
+
+- Running the `substreams run substreams.yaml map_transfers` command will only print the `map_transfers` module's outputs and logs.
+- Running the `substreams run substreams.yaml map_transfers --debug-modules-output=map_pools,map_transfers,store_pools` command will print the outputs of the `map_pools`, `map_transfers`, and `store_pools` modules.
+- Running the `substreams run substreams.yaml map_transfers -s 1000 -t +5 --debug-modules-initial-snapshot=store_pools` command will print all the entries in the `store_pools` module at block 999, then continue with outputs and logs from the `map_transfers` module in blocks 1000 through 1004.
+
+### Parallel Execution for Substreams Modes
+
+The two steps involved during parallel execution are backward execution and forward execution.
```

Comment:
Let's use everywhere `backward parallel execution` and `forward parallel execution`.

```suggestion
The two steps involved during parallel execution are backward parallel execution and forward parallel execution.
```

Reply:
This has been updated through the content. Please let me know if you spot any additional instances. The changes will be available in the next commit/push.